corehq/form_processor/submission_post.py

```diff
@@ -221,12 +195,9 @@ def run(self):
         if failure_response:
             return FormProcessingResult(failure_response, None, [], [], 'known_failures')
 
-        result = process_xform_xml(self.domain, self.instance, self.attachments, self.auth_context.to_json())
+        result = self.form_processor.process_form()
```

Comment:
Alternately, if it is not a concern to have `self.instance` contain different XML than what is used when processing form XML, the ICDS pre-processing hook could possibly be added in `process_xform_xml`?
```suggestion
process_xml = get_form_xml_processor() or process_xform_xml
result = process_xml(self.domain, self.instance, self.attachments, self.auth_context.to_json())
```
with extra import added at the top of this module
```py
from corehq.form_processor.extension_points import get_form_xml_processor
```